tt_eager/tt_dnn/op_library/moreh_arange/moreh_arange_op.hpp

```diff
@@ -21,7 +21,7 @@ struct MorehArange {
     float end;
     float step;
     bool untilize_out;
-    const DataType output_dtype;
+    const std::optional<DataType> output_dtype;
```

Comment:
I have a question.
Is there any reason to use output_dtype as optional? 
I am wondering if there is any problem if I set output_dtype=DataType::BFLOAT16 in `moreh_arange`. :)

Reply:
The reason it was originally optional was that when receiving an output_tensor, if a default value was provided for `output_dtype`, it would be ambiguous to compare it with `output_tensor.get_dtype()`. 

I used optional because I thought there was no `None` in `DataType`, but upon reviewing, I realized there is an `INVALID` value, so this is no longer necessary.